### PR TITLE
[CFM-535-fix] Cancel button action.

### PR DESCRIPTION
**Pull request by anvmn**
https://issuetracker.amplexor.com/jira/browse/CFM-535

### CODE REVIEW

**Comment by anvmn** [comment]
anim

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 1]
```diff
@@ -97,24 +97,21 @@ function c4m_content_form_node_form_alter(&$form, &$form_state, $form_id) {
...
-    $form['actions']['cancel'] = array(
-      '#type' => 'submit',
-      '#value' => t('Cancel'),
-      '#submit' => array('c4m_content_cancel_button_callback'),
-      '#limit_validation_errors' => array(),
-      '#weight' => 14,
+    $form['destination_on_cancel'] = array(
+      // 'name' => 'cancel_destination',
+      '#type' => 'hidden',
+      '#value' => c4m_content_construct_path_to_group_content_page($form['type']['#value']),
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: explain what's in `$form['type']['#value']`

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 2]
```diff
@@ -97,24 +97,32 @@ function c4m_content_form_node_form_alter(&$form, &$form_state, $form_id) {
...
-    );
-  }
-  else {
-    $form['actions']['cancel'] = array(
-      '#type' => 'submit',
-      '#value' => t('Cancel'),
-      '#submit' => array('c4m_content_cancel_button_callback'),
-      '#limit_validation_errors' => array(),
-      '#weight' => 14,
+    $path_tokens = explode('/', rtrim($form['#action'], '/'));
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: @anvmn 
there is a lot of logic in here.
please add examples in an elaborated comment.

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 3]
```diff
@@ -124,6 +131,55 @@ function c4m_content_form_node_form_alter(&$form, &$form_state, $form_id) {
...
+ *
+ * @return string
+ *   Path to group content page
+ *   If $content_type is not group ('group'/'project') or group content,
+ *   defaults to site homepage.
+ */
+function c4m_content_construct_path_to_content_page($content_type) {
+  $group_content_types = array_keys(c4m_og_get_all_group_content_bundles('node'));
+
+  // Content type is not group content.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: this comment doesn't contribute to the understanding of the logic, please fix this.

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 4]
```diff
@@ -124,6 +131,55 @@ function c4m_content_form_node_form_alter(&$form, &$form_state, $form_id) {
...
+    'event' => 'calendar',
+    'news' => 'news',
+  );
+
+  if (in_array($content_type, array_keys($types_path))) {
+    $content_type_path = $types_path[$content_type];
+  }
+  else {
+    // Content type is 'share' - forward to group page.
+    $content_type_path = '';
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: remove this and add `'share' => '',` to the `$types_path`

**Pull request merged**